core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-page-tools-seo/dot-page-tools-seo.component.ts

```diff
@@ -21,21 +29,35 @@ import { DotMessagePipe } from '@dotcms/ui';
     styleUrls: ['./dot-page-tools-seo.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class DotPageToolsSeoComponent implements OnInit {
+export class DotPageToolsSeoComponent implements OnInit, OnChanges {
     @Input() visible: boolean;
-    @Input() currentPageUrl: string;
+    @Input() currentPageUrlParams: DotPageToolUrlParams;
     dialogHeader: string;
     tools$: Observable<DotPageTool[]>;
 
     constructor(private dotPageToolsService: DotPageToolsService) {}
 
     ngOnInit() {
-        this.tools$ = this.dotPageToolsService.get().pipe(
+        this.tools$ = this.getTools();
+    }
+
+    ngOnChanges(changes: SimpleChanges) {
+        if (changes.currentPageUrlParams && !changes.currentPageUrlParams.firstChange) {
+            const prevParams: DotPageToolUrlParams = changes.currentPageUrlParams.previousValue;
+            const currParams: DotPageToolUrlParams = changes.currentPageUrlParams.currentValue;
+            if (prevParams.currentUrl !== currParams.currentUrl) {
+                this.tools$ = this.getTools();
+            }
+        }
+    }
```

Comment:
You are already assigning `tools$` twice, this is the kind of things that can leads to race conditions and unexpected behaviour, I really recommend to add a component store to this component if anything else gets added.